ops/model.py

```diff
@@ -68,7 +70,10 @@
 _StorageDictType = Dict[str, Optional[List['Storage']]]
 _BindingDictType = Dict[Union[str, 'Relation'], 'Binding']
 
-_StatusDict = TypedDict('_StatusDict', {'status': str, 'message': str})
+_SettableStatusName = Literal['active', 'blocked', 'maintenance', 'waiting']
+_SettableStatusNames: Tuple[_SettableStatusName, ...] = get_args(_SettableStatusName)
```

Comment:
I'm a bit uneasy about how similar the two look.
Do you think it would be better, if the constant was _ALL_UPPER_CASED to distinguish it from the _CamelCased type?

Reply:
P.S. I'm also a bit on the fence about the "_Settable" prefix, it seems quirky, like what does "settable" mean in abstract, without reading the rest of the code?

I feel like maybe this is inverted? WDYT about:
- the user-visible type would be called `_StatusName`, and
- the internal union would be called e.g. `_StatusNameOrInternal` or some such?

Reply:
I agree that the name is a bit strange, especially out of context. `_StatusDict` was already used in `model.py` (with `str` typed 'status' key), while `_StatusName` was used only in tests (and contained 5 of the 6 Juju statuses), while the `Settable` prefix is just what I came up with, and I'd definitely be happy to change it to something better.

Perhaps marking the non-settable names explicitly as read-only would improve things. Does this help readability enough, do you think?
```python
_ReadOnlyStatusName = Literal['error', 'unknown']
_SettableStatusName = Literal['active', 'blocked', 'maintenance', 'waiting']
_StatusName = Union[_SettableStatusName, _ReadOnlyStatusName]
_StatusDict = TypedDict('_StatusDict', {'status': _StatusName, 'message': str})
_SETTABLE_STATUS_NAMES: Tuple[_SettableStatusName, ...] = get_args(_SettableStatusName)
```
I'm wary about making the full union name too unfriendly as it reflects the `name` attribute type for the user facing type `StatusBase`, which includes `ErrorStatus` and `UnknownStatus`. While these can't be used to `set` the status without causing an error, you could definitely `get` such a status.

Maybe a suffix is better? Would improve discoverability at least.
```python
_StatusNameReadOnly = Literal['error', 'unknown']
_StatusNameSettable = Literal['active', 'blocked', 'maintenance', 'waiting']
_StatusName = Union[_StatusNameSettable, _StatusNameReadOnly]
_StatusDict = TypedDict('_StatusDict', {'status': _StatusName, 'message': str})
_SETTABLE_STATUS_NAMES: Tuple[_StatusNameSettable, ...] = get_args(_StatusNameSettable)
```

Reply:
I'm not too worried about these, given they're internal (though see my comment elsewhere -- maybe `StatusName` should be public, given that it's the type annotation of a public field).